docs/proposals/33-machine-image-capabilities.md

```diff
@@ -0,0 +1,639 @@
+            communityGalleryImageID: /CommunityGalleries/xzy/Images/gardenlinux-nvme-arm64-gen2/Versions/1592.2.0
+            version: 1592.2.0 # <-- no pre-release tag required anymore
+            #acceleratedNetworking: true # <-- not required anymore
+
+          - capabilities:
+              architecture: [amd64]
+              hypervisorType: ["gen2"]
+            communityGalleryImageID: /CommunityGalleries/xzy/Images/gardenlinux-nvme-gen2/Versions/1592.2.0
+            version: 1592.2.0
+          - capabilities:
+              architecture: [amd64]
+              hypervisorType: ["gen1"]
+            communityGalleryImageID: /CommunityGalleries/xzy/Images/gardenlinux-nvme/Versions/1592.2.0
+            version: 1592.2.0
+```
+
+The algorithm to determine if an image is valid for a machine type is given as follows: 
+
+For every capability the union of values that is supported by the machine type and the machine image must not be empty.
+In other words, the machine image must support at least one value of each capability of the machine type.
+
+```js
+for capabilityName, machineCapabilities in machineType.capabilities:
+  if union(imageVersion.capabilities[capabilityName], machineCapabilities) is empty:
+    return false
+      
+return true
+```
+
+## Implications on NamespacedCloudProfiles. 
+
+The following implications are to be considered in regards to `NamespacedCloudProfile`s:
+ 
+* `NamespacedCloudProfile`s won't have a global `capabilities` definition.
+* For overridden machine image versions, no `capabilities` must be defined, as they are inherited from the machine image versions of the parent `CloudProfile`.
+* For custom machine image versions, the `capabilities` need to be defined in the `NamespacedCloudProfile`, as they would be in the `CloudProfile` (as well as for the `providerConfig`).
+   
+* This has also implications on later-added equal machine image versions to the parent `CloudProfile`: On conflicting definition, the machine image version of the `NamespacedCloudProfile` will be rendered in the resulting status (see [this PR](https://github.com/gardener/gardener/pull/11093) for reference).
+
+
+### Migration Strategy
+
+Capabilities are **OPT-IN** and not all infrastructure providers or Gardener installations might use them.
+All API changes in Gardener core are backwards compatible.
+They can be implemented first including its filter and admission logic.
+In a second step each provider extension can be updated to include the capabilities in the cloud profile.
+In a third wave cloud profiles and the Gardener Dashboard can be updated to react to the capabilities.
+
+❗️❗️❗️ when the architecture field is marked as deprecated the eventual removal in the  CloudProfile API will be a breaking change. Latest this point all cloud profiles must be updated to include the architecture in the capabilities. It will no longer be opt in.❗️❗️❗️
```

Comment:
```suggestion
> [!IMPORTANT]
> When the `architecture` field is marked as deprecated the eventual removal in the `CloudProfile` API will be a breaking change. Latest at this point all `CloudProfile`s must be updated to include the architecture in the capabilities. It will no longer be opt in.
```